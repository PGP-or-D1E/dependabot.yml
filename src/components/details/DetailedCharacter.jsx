import React from 'react';
import propTypes from 'prop-types';
import styles from './detailedcharacter.css';

const DetailedCharacter = ({ name, image, id }) => {
    
  return (
    <div id={id} className={styles.Detailed}>
      <figure data-testid="one-character">
        <img src={image} />
        <figcaption>{name}</figcaption>
      </figure>
    </div>
  );
};

DetailedCharacter.propTypes = {
  id: propTypes.string.isRequired,
  name: propTypes.string.isRequired,
  image:propTypes.string.isRequired
};

export default DetailedCharacter;
