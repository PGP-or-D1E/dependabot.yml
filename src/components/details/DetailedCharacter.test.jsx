
import React from 'react';
import { render } from '@testing-library/react';
import DetailedCharacter from './DetailedCharacter';

describe('DetailedCharacter Snapshot', () => {
  it('matches a snapshot', () => {
    const { asFragment } = render(<DetailedCharacter   image="http://image.com"
      name="Billy"
      id="fnrivnvkfmnfgknvdfn436"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
