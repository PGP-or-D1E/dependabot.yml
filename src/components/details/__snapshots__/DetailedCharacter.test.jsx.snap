// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedCharacter Snapshot matches a snapshot 1`] = `
<DocumentFragment>
  <div
    class="Detailed"
    id="fnrivnvkfmnfgknvdfn436"
  >
    <figure
      data-testid="one-character"
    >
      <img
        src="http://image.com"
      />
      <figcaption>
        Billy
      </figcaption>
    </figure>
  </div>
</DocumentFragment>
`;
